mesonbuild/modules/i18n.py

```diff
@@ -62,9 +60,7 @@ class Gettext(TypedDict):
     class ItsJoinFile(TypedDict):
 
         input: T.List[T.Union[
-            str, build.BuildTarget, build.CustomTarget, build.CustomTargetIndex,
-            build.ExtractedObjects, build.GeneratedList, ExternalProgram,
-            mesonlib.File]]
+            str, mesonlib.File, build.CustomTarget, build.CustomTargetIndex, build.GeneratedList]]
```

Comment:
Passing `ExternalProgram` or `ExtractedObjects` results in an unhandled exception before this PR.

As to why I also removed build.BuildTarget:
The argument will be passed to msgfmt or itstool eventually.

These tools accept *.mo, *.xml and *.desktop files. (They don't check the filename, but just care about the content.) I don't see a way how a BuildTarget could also be a *valid and useful* *.mo or *.xml or *.desktop file.

If you pass an ELF file to msgfmt, it will print warnings, and output an empty file.